mlflow/pyspark/ml/__init__.py

```diff
@@ -643,19 +908,81 @@ def fit_mlflow(original, self, *args, **kwargs):
             _logger.warning(_get_warning_msg_for_fit_call_with_a_list_of_params(self))
             return original(self, *args, **kwargs)
         else:
-            estimator = self.copy(params)
+            # we need generate estimator param map so we call `self.copy(params)` to construct
+            # an estimator with the extra params.
+            estimator = self.copy(params) if params is not None else self
             _log_pretraining_metadata(estimator, params)
             spark_model = original(self, *args, **kwargs)
             _log_posttraining_metadata(estimator, spark_model, params)
             return spark_model
 
     def patched_fit(original, self, *args, **kwargs):
+        metrics_manager = _get_autologging_metrics_manager()
+        should_log_post_training_metrics = metrics_manager.should_log_post_training_metrics()
         with _SparkTrainingSession(clazz=self.__class__, allow_children=False) as t:
             if t.should_log():
-                return fit_mlflow(original, self, *args, **kwargs)
+                with metrics_manager.disable_log_post_training_metrics():
+                    fit_result = fit_mlflow(original, self, *args, **kwargs)
+                # In some cases the `fit_result` may be an iterator of spark models.
```

Comment:
When does `fit` return an iterator?

Reply:
This case:
https://github.com/apache/spark/blob/7d823367348c0235ffb212b09afcf053d070068d/python/pyspark/ml/base.py#L156